sdk/servicebus/service-bus/src/core/receiverHelper.ts

```diff
@@ -82,4 +105,8 @@ export class ReceiverHelper {
 
     return drainPromise;
   }
+
+  private _isValidReceiver(receiver: Receiver | undefined): receiver is Receiver {
+    return receiver != null && receiver.isOpen();
```

Comment:
I see `isRemoteOpen` too in rhea-promise.
![image](https://user-images.githubusercontent.com/10452642/89583621-81058680-d7ef-11ea-8c8e-f17b68d99261.png)

When should we check for that?

Reply:
Hm...quick search of the code base doesn't show that we're using that flag (aside from at the connection level to check if the connection is that weird "1 second pause" state before fully opening).

Might be a larger question to answer than I want to do in this PR. I wonder if we can use some of the mockhubs stuff to test it out.

Reply:
Yes, not relevant to this PR, just wanted to ignite some feelings about this.

Reply:
It was a great point and we should look into it. Filed as #10482